fix: Use config.appPath allows finding plugins when sk server is required/imported

webmasterkai opened the pull request:
See #555. If signalk server is added to another project with a `require` or `import` the relative path does not correctly find plugins. Using `config.appPath` does work however. This makes that change and includes some cleanup work by @taggie05.

webmasterkai commented:
Ping @sbender9 or @tkurki. It's been several months without input/feedback. I need this for several projects I'm working on.

tkurki commented on lib/modules.js:
```diff
@@ -22,33 +22,27 @@ const path = require('path')
-  return res
+  // look in configPath for modules
+  if (app.config.configPath !== app.config.appPath) {
+    searchAddModules(app.config.configPath)
```
For some reason `searchAddModules` is not very intuitive for me. Would it make sense that this line would read `findModules(app.config.configPath, keyword).forEach(m => modules.add(m))` ? Then `findModules(dir, keyword)` could be static, no side effects.

Mergeable as it is, this is no biggie.

webmasterkai commented:
I removed `searchAddModules()` completely.

`modulesWithKeyword()` is now much more functional in style. No side effects anywhere. `getModulePaths()` has been added to extract and path strings from `app.config` object. I added more comments to help explain what's going on.

I also included a note and link to docs that I'm using the functional version of lodash [lodash/fp](https://github.com/lodash/lodash/wiki/FP-Guide).

I know you usually ask about squashing commits. Is that something that should be made a standard practice by updating the [https://github.com/SignalK/signalk-server-node/blob/master/docs/CONTRIBUTING.md

tkurki commented:
Oh. You rewrote the whole thing - I rather liked the previous version better with the Sets. I find it a lot easier to understand than this one - we have not used `lodash/fp`, except in your contributions. 

I personally have had several instances in the Real World where `lodash/fp` was used in an another not-very-functional codebase and developers joining the project after fact struggled - for very little gain in readability or performance.

While elegantly functional I find it hard to reason what is happening here with my almost nonexistent lodash/fp experience. Two passes with the Set is immediately obvious.

So....I'd like to go back, if you don't mind.

As for squashing: I think it is easier for me to just press the `Squash and merge` button in Github than expect people to have nice commit histories. Most PRs are so small that if there are multiple commits the history is of the "changethis-fix1-changemore-fix2-..." variety where it would need a complete rewrite to be meaningful instead of just squashing.
